neml/drivers.py

```diff
@@ -596,7 +596,212 @@ def strain_cyclic(model, emax, R, erate, ncycles, T = 300.0, nsteps = 50,
+def strain_cyclic_extrapolated(model, emax, R, erate, ncycles, T = 300.0, nsteps = 50,
+    sdir = np.array([1,0,0,0,0,0]), hold_time = None, n_hold = 25,
+    verbose = False, check_dmg = False, dtol = 0.75, min_cycle=3, unit_extrapolate = 10,
+    jump_del_N=10,allowable_jump_stress=5.0):
+  """
+    Strain controlled cyclic test extrapolation.
+
+    Extra Keyword Args:
+      min_cycle                minimum cycles to start the extrapolation process
+      unit_extrapolate         number of cycles to perform single cycle extrapolation
+      jump_del_N               number of cycles to jump
+      allowable_jump_stress    extrapolate when stress jump is within this limit
+
+    Returns:
+      dict:         results dictionary containing...
+
+    **Results in dictionary:**
+      ============= ========================
+      Name          Description
+      ============= ========================
+      cycles:       list of cycle numbers
+      max:          maximum stress per cycle
+      min:          minimum stress per cycle
+      ============= ========================
+  """
+  # Setup
+  driver = Driver_sd(model, verbose = verbose, T_init = T)
+  emin = emax * R
+  if hold_time:
+    if np.isscalar(hold_time):
+      hold_time = [hold_time, hold_time]
+  else:
+    hold_time = [0,0]
+
+  # Setup results
+  strain = [0.0]
+  stress = [0.0]
+  time = [0.0]
+  cycles = []
+  smax = []
+  smin = []
+  smean = []
+
+  ecycle = []
+  pcycle = []
+
+  # First half cycle
+  if verbose:
+    print("Initial half cycle")
+  e_inc = emax / nsteps
+  try:
+    for i in range(nsteps):
+      if i == 0:
+        einc, ainc = driver.erate_einc_step(sdir, erate, e_inc, T)
+      else:
+        einc, ainc = driver.erate_einc_step(sdir, erate, e_inc, T, einc_guess = einc,
+            ainc_guess = ainc)
+      if check_dmg:
+        if driver.stored_int[-1][0] > dtol:
+          raise Exception("Damage check exceeded")
+
+      strain.append(np.dot(driver.strain_int[-1], sdir))
+      stress.append(np.dot(driver.stress_int[-1], sdir))
+      time.append(time[-1] + e_inc / erate)
+
+  except Exception as e:
+    print("Failed to make first half cycle")
+    raise e
+  s = 0
+
+# steps in one cycle
+  if (hold_time[0] > 0) and (hold_time[1] == 0):
+      steps = 2*nsteps + n_hold
+  elif (hold_time[1] > 0) and (hold_time[0] == 0):
+      steps = 2*nsteps + n_hold
+  elif (hold_time[0] > 0) and (hold_time[1] > 0):
+      steps = 2*nsteps + 2*n_hold
+  else:
+      steps = 2*nsteps
+
+  extrapolate = False
+  while s < ncycles:
+    if verbose:
+      print("Cycle %i" % s)
+
+    if check_dmg:
+      if driver.stored_int[-1][0] > dtol:
+        print("Damage check exceeded")
+        break
+
+    if (s >= min_cycle) and (extrapolate == True):        # No extrapolation before min_cycle
+        if (s <= unit_extrapolate):                       # single cycle jump for first unit_extrapolate cycles
+            del_N = 1
```

Comment:
same as above, `del` is confusing me, I don't know if you are deleting something... either `N` or `delta_N`.

Reply:
changed to delta